sklearn/base.py

```diff
@@ -237,6 +206,54 @@ def set_params(self, **params):
 
         return self
 
+
+class BaseEstimator(GetSetParamsMixin):
+    """Base class for all estimators in scikit-learn
+
+    Notes
+    -----
+    All estimators should specify all the parameters that can be set
+    at the class level in their ``__init__`` as explicit keyword
+    arguments (no ``*args`` or ``**kwargs``).
+    """
+
+    def get_params(self, deep=True):
+        """Get parameters for this estimator.
+
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        try:
+            return super().get_params(deep=deep)
+        except RuntimeError as e:
+            raise RuntimeError("scikit-learn estimators should always "
+                               "specify their parameters in the signature"
+                               " of their __init__ (no varargs)."
+                               " %s doesn't follow this convention."
+                               % self.__class__) from e
+
+    def set_params(self, **params):
```

Comment:
Left to preserve the docstring.

Reply:
I would just use the inherited method instead of overriding, but I left it as I wasn't sure if there are any rules regarding docstrings.